### PR TITLE
At/cb233/implement the add series modal

### PR DESCRIPTION
**Pull request by Angelbt12**
image

### CODE REVIEW

File: src/components/modals/AddSeriesModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  StyledContainerClose,
+  StyledContainerTitle,
+} from "./styles";
+
+interface FormValues {
+  field1: number;
+  field2: number;
+}
+
+export interface Option {
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:23]: Relocate interface in type.ts file

File: src/components/modals/AddSeriesModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+  const node = document.getElementById(portalId);
+  if (!node) {
+    throw new Error(
+      "The portal node is not defined. This can occur when the specific node used to render the portal has not been defined correctly."
+    );
+  }
+
+  const [form, setForm] = useState({
+    paymentMethod: "",
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:61]: Create interface

File: src/components/modals/AddSeriesModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    field1: "",
+    field2: "",
+  });
+
+  const onChange = (name: string, newValue: string) => {
+    setForm({ ...form, [name]: newValue });
+  };
+
+  const handleNumberChange = (name: string, value: string) => {
+    const numericValue = value.replace(/[^0-9]/g, "");
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:73]: For validations you can implement Formik

File: src/components/modals/AddSeriesModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+  return createPortal(
+    <Blanket>
+      <StyledModal $smallScreen={isMobile}>
+        <StyledContainerTitle>
+          <Text type="headline" size="small">
+            {title}
+          </Text>
+          <StyledContainerClose onClick={handleClose}>
+            <Stack alignItems="center" gap={inube.spacing.s100}>
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:119]: Design-system space should not be implemented

File: src/components/modals/AddSeriesModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+              <Text>Cerrar</Text>
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:120]: Texts should be left in config

File: src/components/modals/AddSeriesModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+            required={false}
+            size="wide"
+            value={form["paymentMethod"]}
+          />
+          <Textfield
+            id="field1"
+            label="Cantidad"
+            placeholder="Número de pagos"
+            fullwidth
+            type="text"
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:149]: The type must be number

File: src/components/modals/AddSeriesModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+          onChange={handleDateChange}
+          onBlur={handleDateBlur}
+          onFocus={handleDateFocus}
+          message="Formato de fecha no válido"
+          status={form.firstPayment.status}
+          size="wide"
+          fullwidth
+        />
+        <Divider />
+        <Stack justifyContent="flex-end" margin="s200 s0" gap="10px">
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/index.tsx:194]: Invalid margin value

File: src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import { useState } from "react";
+import { StoryFn, Meta } from "@storybook/react";
+import { Button } from "@inube/design-system";
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx:3]: Do not use inube/design-system

File: src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  { id: "monthly", label: "Mensual", value: "monthly" },
+  { id: "quarterly", label: "Trimestral", value: "quarterly" },
+  { id: "annually", label: "Anual", value: "annually" },
+];
+
+const story: Meta<typeof AddSeriesModal> = {
+  component: AddSeriesModal,
+  title: "components/modals/AddSeriesModal",
+  argTypes: props,
+};
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx:27]: Missing add parameters property for general component documentation

File: src/components/modals/AddSeriesModal/styles.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import styled from "styled-components";
+import { inube } from "@inube/design-system";
```
Comment (diegoAndres01) [inline src/components/modals/AddSeriesModal/styles.ts:2]: Do not use inube/design-system

**Review by diegoAndres01 — CHANGES_REQUESTED** [review]

File: src/components/modals/AddSeriesModal/index.tsx (hunk 1 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/index.tsx:23]: ok

File: src/components/modals/AddSeriesModal/index.tsx (hunk 5 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/index.tsx:120]: ok

File: src/components/modals/AddSeriesModal/index.tsx (hunk 6 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/index.tsx:149]: ok

File: src/components/data/TableExtraordinaryInstallment/Detail/index.tsx [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import { Icon } from "@inubekit/icon";
 import { useState } from "react";
-import { MdOutlineMoreVert } from "react-icons/md";
+import { MdOutlineEdit , MdDeleteOutline} from "react-icons/md";
 import { ActionModal } from "../Actions";
 import { Stack } from "@inubekit/stack";
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/data/TableExtraordinaryInstallment/Detail/index.tsx:5]: ok

File: src/components/modals/AddSeriesModal/index.tsx (hunk 7 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/index.tsx:157]: ok

File: src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx (hunk 8 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/stories/PromissoryNotesModal.stories.tsx:3]: ok

File: src/components/modals/AddSeriesModal/styles.ts (hunk 10 above)
Comment (Angelbt12) [reply src/components/modals/AddSeriesModal/styles.ts:2]: ok

File: src/components/data/TableExtraordinaryInstallment/Detail/index.tsx [hunk 12]
```diff
@@ -15,15 +15,26 @@ export function Detail(props: DetailProps) {
...
-        icon={<MdOutlineMoreVert />}
-        size="12px"
-        cursorHover
-        appearance="primary"
-        onClick={() => setModalOpen(true)}
-        shape="circle"
-        variant="filled"
-      />
+      <Stack padding="0px 16px">
+        <Icon
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/data/TableExtraordinaryInstallment/Detail/index.tsx:19]: ok

File: src/components/ErrorAlert/styles.ts [hunk 13]
```diff
@@ -49,15 +49,14 @@ const slideDown = keyframes`
 
 export const AlertContainer = styled.div`
   display: flex;
-  align-items: center;
   background-color: ${({ theme }) =>
     theme?.color?.stroke?.gray?.regular || inube.color.surface.warning.clear};
   padding: 7px;
   border-radius: 4px;
   border: 1px solid
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/ErrorAlert/styles.ts:56]: ok

File: src/components/ErrorAlert/config.ts [hunk 14]
```diff
@@ -1,4 +1,3 @@
-// src/config.ts
 export const messages = {
     defaultError: "Existe un error sin evaluar",
```
[the remark this answers is not in the document]
Comment (Angelbt12) [reply src/components/ErrorAlert/config.ts]: ok

**Review by carin91 — APPROVED** [review]

**Pull request merged**